Assets/Scripts/Models/Area/World.cs

```diff
@@ -338,6 +333,74 @@ public void ReadJson(string filename)
         tileGraph = new Path_TileGraph(this);
     }
 
+    public void ResizeWorld(Vector3 worldSize)
+    {
+        ResizeWorld((int)worldSize.x, (int)worldSize.y, (int)worldSize.z);
+    }
+
+    public void ResizeWorld(int width, int height, int depth)
+    {
+        if (width < Width || height < Height || depth < Depth)
+        {
+            if (width < Width)
+            {
+                Debug.LogWarning("Width too small: " + Width + " " + width);
+            }
+
+            if (height < Height)
+            {
+                Debug.LogWarning("Height too small: " + Height + " " + height);
+            }
+
+            if (depth < Depth)
+            {
+                Debug.LogWarning("Depth too small: " + Depth + " " + depth);
+            }
+
+            Debug.LogError("Shrinking the world is not presently supported");
+            return;
+        }
+
+        if (width == Width && height == Height && depth == Depth)
+        {
+            // No change, just bail
+            return;
+        }
+
+        int offsetX = (width - Width) / 2;
+        int offsetY = ((height - Height) / 2) + 1;
+
+        Tile[,,] oldTIles = (Tile[,,])tiles.Clone();
```

Comment:
What's with the funky variable name? Your scaring me here with both capital i and l together.